### PR TITLE
CDPS-883: Update node.js to v20.16 and upgrade non-major dependencies

### PR DESCRIPTION
**Pull request by brightonsbox**
The minor updates introduced loads of typescript compilation failures complaining where object values were accessed via  `object[key]` syntax, hence a load of random tidy up as part of this PR.  It did actually uncover a minor bug, and it's encouraged stronger typing

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,4 +1 @@
-#!/bin/sh
-. "$(dirname "$0")/_/husky.sh"
-
```
Comment (brightonsbox) [inline .husky/pre-commit:3]: After noticing a deprecation notice after husky upgrade, I've followed the process here https://github.com/typicode/husky/releases/tag/v9.0.1

File: package.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "repository": "git@github.com:ministryofjustice/hmpps-prisoner-profile.git",
   "license": "MIT",
   "scripts": {
-    "prepare": "husky install",
```
Comment (brightonsbox) [inline package.json:8]: Again part of release 9...

File: server/services/professionalContactsService.ts [hunk 3]
```diff
@@ -209,7 +209,7 @@ export default class ProfessionalContactsService {
     }
 
     // Return the most recently created record for each of the YOI contact relationships if available
-    sortArrayOfObjectsByDate<Contact>(contacts.otherContacts, 'createdDateTime', SortType.ASC).forEach(c => {
+    sortArrayOfObjectsByDate<Contact>(contacts.otherContacts, 'createDateTime', SortType.ASC).forEach(c => {
```
Comment (brightonsbox) [inline server/services/professionalContactsService.ts:212]: Stronger typing flagged that this was the wrong key

File: server/mappers/headerMapper.test.ts [hunk 4]
```diff
@@ -30,13 +30,13 @@ describe('HeaderMapping', () => {
...
-      expect(headerData['photoType']).not.toBeDefined()
-      expect(headerData['restrictedPatient']).not.toBeDefined()
-      expect(headerData['hideBanner']).not.toBeDefined()
+      expect(headerData['backLinkLabel' as keyof typeof headerData]).not.toBeDefined()
+      expect(headerData['profileBannerTopLinks' as keyof typeof headerData]).not.toBeDefined()
+      expect(headerData['alerts' as keyof typeof headerData]).not.toBeDefined()
+      expect(headerData['tabLinks' as keyof typeof headerData]).not.toBeDefined()
+      expect(headerData['photoType' as keyof typeof headerData]).not.toBeDefined()
+      expect(headerData['restrictedPatient' as keyof typeof headerData]).not.toBeDefined()
+      expect(headerData['hideBanner' as keyof typeof headerData]).not.toBeDefined()
```
Comment (whitfield-mj) [inline server/mappers/headerMapper.test.ts:39]: Why can't these just use the dot syntax like the ones above?

**Review by whitfield-mj — APPROVED** [review]

File: server/mappers/headerMapper.test.ts (hunk 4 above)
Comment (brightonsbox) [reply server/mappers/headerMapper.test.ts:39]: Yeah bit of a strange test... these fields don't exist on the object - I think it's trying to show that it's definitely not the other object that has these fields being returned 🤷

**Pull request merged**